src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -22,6 +23,90 @@ namespace System
     /// </summary>
     public static class Span
     {
+        /// <summary>
+        /// Determines whether the beginning of the span matches the specified value when compared using the specified comparison option.
+        /// </summary>
+        public static bool StartsWith(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparison)
+        {
+            if ((uint)(comparison - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
+            {
+                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparison));
+            }
+
+            if (span.Length < value.Length)
+            {
+                return false;
+            }
+
+            switch (comparison)
+            {
+                case StringComparison.CurrentCulture:
+                    return CultureInfo.CurrentCulture.CompareInfo.IsPrefix(span, value, CompareOptions.None);
+
+                case StringComparison.CurrentCultureIgnoreCase:
+                    return CultureInfo.CurrentCulture.CompareInfo.IsPrefix(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.InvariantCulture:
+                    return CompareInfo.Invariant.IsPrefix(span, value, CompareOptions.None);
+
+                case StringComparison.InvariantCultureIgnoreCase:
+                    return CompareInfo.Invariant.IsPrefix(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.Ordinal:
+                    // TODO: https://github.com/dotnet/corefx/issues/25182
+                    // return span.StartsWith(value);
+                    throw new NotImplementedException();
+
+                case StringComparison.OrdinalIgnoreCase:
+                    return CompareInfo.CompareOrdinalIgnoreCase(span, value) == 0;
```

Comment:
This isn't correct, is it?

Reply:
Steve is correct. this comparison is wrong. you should slice the span with the value length. 

for value.Length === 0 this check should be performed early for all type of comparisons and we should return true at that time.

---
In reply to: [166247652](https://github.com/dotnet/coreclr/pull/16223#discussion_r166247652) [](ancestors = 166247652)